specs/fault-proof/stage-one/anchor-state-registry.md

```diff
@@ -349,10 +367,68 @@ finalize withdrawals with invalidated games would be considered Critical Severit
 
 #### Dependencies
 
-- [aASR-003](#aasr-003-optimismportal-properly-reports-respected-game-type-blacklist-and-retirement-time)
+- [aASR-003](#aasr-003-incorrectly-resolving-games-will-be-invalidated-before-they-have-valid-claims)
 
 ## Function Specification
 
+### constructor
+
+- MUST set the value of the [Dispute Game Finality Delay](#dispute-game-finality-delay-airgap).
+
+### initialize
+
+- MUST only be triggerable once.
+- MUST set the value of the `SuperchainConfig` contract that stores the address of the Guardian.
+- MUST set the value of the `DisputeGameFactory` contract that creates Dispute Game instances.
+- MUST set the value of the [Starting Anchor State](#starting-anchor-state).
+- MUST set the value of the initial [Respected Game Type](#respected-game-type).
+- MUST set the value of the [Retirement Timestamp](#retirement-timestamp) to the current block
+  timestamp. NOTE that this is a safety mechanism that invalidates all existing Dispute Game
+  contracts to support the safe transition away from the `OptimismPortal` as the source of truth
+  for game validity. In this way, the `AnchorStateRegistry` does not need to consider the state of
+  the legacy blacklisting/retirement mechanisms within the `OptimismPortal` and starts from a clean
+  slate.
```

Comment:
Are we moving away from the re-initialization pattern we use for upgrades (i.e. using `StorageSetter` to clear `initialized`, and then re-calling `initialize`?) With this, invalidating ongoing withdrawals could be a footgun for ASR proxy upgrades if not.

Reply:
My understanding is that future upgrades will need a separate `upgrade` function. Let me ping @maurelian to confirm.

Reply:
Sounds good. Can resolve this comment once this is sorted; I don't think this is too bad, just need to make sure the potential footgun is documented. Even if it was hit, though, it'd be safe (just a temp withdrawal liveness failure.)

Reply:
@smartcontracts is correct that the way we'll handle adding new state vars in the future is with a `function(uint newVar) public initializer()` method that only sets the new variables. 

However in this case I believe the intent is for this to be an entirely new superchain-wide ASR(?), so it's not being upgraded from a previous state and wouldn't need that.